routes/index.js

```diff
@@ -18,3 +19,17 @@ router.get("/rankings", (req, res) => {
 });
 
 module.exports = router;
+
+
+//to check for visibility of a question and then print it on to the required page.
+const ProblemsModel = require("../models/problems");
+function finder() { 
+        ProblemsModel.find({isVisible: 1}).then(function(doc) {
+        var table = [];
+        for(var i=0; i < doc.length ; i++ )
+        {
+            table.push([doc[i].id,doc[i].name,doc[i].tags,doc[i].difficulty]);
+        }
```

Comment:
@dipanshu231099  There's no need to run the loop and doing so much redundant work. You can easily find `doc` where **isVisible:1** and have fields **qID, name, tags, difficulty** and sorted in **descending order** of qID. First, do these changes and bring the correct output by console.log(doc). Then we'll come to passing this to the webpage.

Reply:
also @dipanshu231099 try squashing your commits to clear your commit history.. each significant addition should be in a single commit . Lookup online..

Reply:
@dipanshu231099 Making it a little easier for you, make it all a single commit. https://github.com/servo/servo/wiki/Beginner's-guide-to-rebasing-and-squashing

Reply:
@Vishal1541 sir maine pehle module.find(....).then(doc => console.log(doc)) hi use kiya tha aur usse saara database bhi sahi aa rha tha

Reply:
par mujhe problem aa rhi hai passing it to the webpage

Reply:
ProblemsModel.find({isVisible: 1}).then(doc => res.render("problem_set",{table: doc}));
i have also tried this line

Reply:
Sahi nhi aa raha. You are not sorting as required. And remove the for loop from there. The whole database thing which has been asked can be done in a single line. Try googling. First complete this then we'll jump to passing the data